models/resultModel.js

```diff
@@ -0,0 +1,46 @@
+const mongoose = require('mongoose');
+
+const answerSchema = new mongoose.Schema(
+  {
+    body: String,
+    weight: Number,
+    choosen: {
+      type: Boolean,
+    },
+  },
+  { timestamps: true }
+);
+
+const questionSchema = new mongoose.Schema(
+  {
+    body: String,
+    answers: [answerSchema],
+  },
+  { timestamps: true }
+);
+
+const scoreSchema = new mongoose.Schema(
+  {
+    min: Number,
+    max: Number,
+    result: String,
+    description: String,
+  },
+  { timestamps: true }
+);
+
+const resultSchema = new mongoose.Schema(
+  {
+    user: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
+    tags: [String],
+    questions: [questionSchema],
+    scores: [scoreSchema],
```

Comment:
I don't mind having a duplicate schema for the result.
But in this way, you don't have the score out of the box, you have to calculate it every time. 
Instead, it's better to have a field for the user's final score, and the scores array is kind of redundant to have all scores listed. we're only interested in the ones with (min < user's final score < max) [and that is only if the questionnaire is testing for a lot of conditions at the same time i.e., Depression, anixity, and stress. This rarely happens]

Reply:
a final word on this matter. On postman, you added the whole object with the createdAt, UpdatedAt, and _id fields for creating a result, which is not quite accurate and might confuse the API consumers.